core/domain/exp_services.py

```diff
@@ -760,9 +761,11 @@ def delete_explorations(committer_id, exploration_ids, force_deletion=False):
     # Delete the explorations from search.
     search_services.delete_explorations_from_search_index(exploration_ids)
 
-    # Delete the exploration summaries, regardless of whether or not
-    # force_deletion is True.
+    # Delete the exploration summaries, recommendations and opportunities
+    # regardless of whether or not force_deletion is True.
     delete_exploration_summaries(exploration_ids)
+    recommendations_services.delete_exploration_recommendations(exploration_ids)
```

Comment:
Do we currently have old models of this form that we should also be cleaning up?

Reply:
Done, now we remove all the previous `ExplorationRecommendationsModel`s when `CronExplorationRecommendationsHandler` is run.